x/concentrated-liquidity/types/keys.go

```diff
@@ -94,8 +94,9 @@ func KeyPool(poolId uint64) []byte {
 	return []byte(fmt.Sprintf("%s%d", PoolPrefix, poolId))
 }
 
-func KeyIncentiveRecord(poolId uint64, denom string, minUptime time.Duration) []byte {
-	return []byte(fmt.Sprintf("%s%s%d%s%s%s%d", IncentivePrefix, KeySeparator, poolId, KeySeparator, denom, KeySeparator, uint64(minUptime)))
+func KeyIncentiveRecord(poolId uint64, denom string, minUptime time.Duration, addr sdk.AccAddress) []byte {
+	addrKey := address.MustLengthPrefix(addr.Bytes())
```

Comment:
I noticed similar formatting while working on positions and ended up removing them due to the edge cases that stem from the prefix: https://github.com/osmosis-labs/osmosis/pull/4602

Why do we need it? If we are surrounding an address with a key separator, there is no need to know the length.

The length should only be needed when we are unsure how to deserialize the address from the store. Given that we have key separator placeholders around, we should be able to write the address directly to the store